### DISCUSSION TITLE
Getting regulon output table

### DISCUSSION THREAD

**Discussion by bvaldebenitom** [aertslab/SCENIC#390] [Python]
Hi everyone!

I've been going back through some pyscenic commands I ran a few weeks ago, and in the output generated I have one like this:
```
Regulon,target,importance
AU041133(+),0.02598072304253577,0.014035722159488333
Alx4(+),0.030191280157972507,0.018425783808366573
Arnt(+),0.02492186993283134,0.011383900828532648
Arntl(+),0.024480617823556892,0.02384036412217325
Atf1(+),0.02411312362195015,0.013940085570176928
Atf3(+),0.023916541488436787,0.033444611270576706
Atf4(+),0.024601773562979365,0.03014477022321595
Atf5(+),0.026324994422788465,0.03161108411113646
Atf6(+),0.024517557899016343,0.009507088798893113
Atf7(+),0.02606789793728184,0.018307109625973643
```

It has a total of 362 lines. Now, re-running the `pyscenic aucell` command, I got a CSV matrix, with the same amount of lines. I'm not sure if I missed something now. The command I'm using is:

`pyscenic aucell --output aucell_output.csv --transpose --num_workers 50 --seed 777 count_matrix.csv regulons.csv`

where` count_matrix.csv `is the single cell experiment raw count matrix, and `regulons.csv` is the output from the `ctx `step:
```
,,Enrichment,Enrichment,Enrichment,Enrichment,Enrichment,Enrichment,Enrichment,Enrichment
,,AUC,NES,MotifSimilarityQvalue,OrthologousIdentity,Annotation,Context,TargetGenes,RankAtMax
TF,MotifID,,,,,,,,
Arnt,flyfactorsurvey__tgo_sima_SANGER_5_FBgn0015014,0.06233425804923157,3.023942464692646,0.0,0.479751,motif is annotated for orthologous gene FBgn0264075 in D. melanogaster (identity = 47%),"frozenset({'weight>75.0%', 'mm10__refseq-r80__10kb_up_and_down_tss.mc9nr.genes_vs_motifs.rankings', 'activating'})","[('Slc31a1', 0.7181839055285076), ('Flrt2', 0.3974843591197314), ('Tppp3', 1.0224520561082902), ('Dhrs3', 2.504602898090744), ('Asxl1', 1.4503746173829155), ('Luc7l2', 2.627619203659232), ('Ubap2l', 0.6646154543799697), ('Rab27a', 0.6287652719580377), ('Trim8', 1.9371294842203093), ('Tbc1d4', 0.9715612356506548), ('Sertad4', 0.7114292775862248), ('Prickle2', 0.5718744417941558), ('Tnrc18', 0.4076175313888652), ('Arid1b', 4.188702249407746), ('Chrm2', 0.5306040670334597), ('Mxi1', 0.797051014581809), ('Phf21a', 0.5540405029888396), ('Trps1', 0.5316463205051527), ('Ccnl1', 0.7105056300207041), ('Rere', 1.580672671599915), ('Slc1a3', 0.4137081985295083), ('Dnmbp', 0.5140944482840842), ('Scube3', 2.0985761829455662), ('Ep300', 0.5660203226539599), ('Kansl1', 1.56891047492277), ('Gabarapl1', 0.676036379551916)]",390
```

Any input or directions as to whether there is something missing in my command, or which step is the one generating the regulon table, will be highly appreciated!

Thanks.

**Comment by Tripfantasy** [comment]
Hi! Just happened to stumble across this. This table seems odd. Comparing to my outputs, the column "target" should be a gene name. The "importance" column seems to be a measure of the impact of a certain TF on the target gene. I was able to reproduce this at TF level by examining the output from the grn step. (adjacencies.csv) 

I would check this out if you want a general idea of what some of these files look like. They did a good job showing the desired structure in their workflow:

http://htmlpreview.github.io/?https://github.com/aertslab/SCENICprotocol/blob/master/notebooks/PBMC10k_downstream-analysis.html
